### DISCUSSION TITLE
[Feature] Show Face Recognition for Partner

### DISCUSSION THREAD

**Discussion by neuleo** [immich-app/immich#5089] [Feature Request]
### The feature

Hi, it would be nice that when a user shares his gallery, the other user can also use the face recognition feature. So that the names are then displayed. 

It would also be nice if you could also see the pictures of your partner on the map. It would be nice to have an option that you set once and that is then saved for the future.

### Platform

- [X] Server
- [X] Web
- [X] Mobile

**Comment by ldivl** [comment]
Yes when sharing album, or partner sharing it should include map and faces which are already tagged.  I’ve been using shared login for now.

**Comment by Kaaybi** [comment]
Just for information, there was an issue about this: https://github.com/immich-app/immich/issues/2478
Also might be a duplicate to https://github.com/immich-app/immich/discussions/6024

**Comment by risingPhil** [comment]
Just for the record: the tickets referenced by Kaaybi are closed already. (for duplication reasons I suppose)

So it appears this is the last remaining ticket.

And yes, I'm also interested in this feature. The lack of it makes the partner sharing feature feel incomplete. (another missing feature is the fact that I can't give permissions to her to delete photos from my account and vice versa) In fact, if we'd both use the same account it would give us a better experience. So that kinda defeats the point of having user accounts in the first place.

**Comment by ttyridal** [comment]
pull request #7045  :)

**Reply by ttyridal** [reply]
Though, that PR is "read-only", so typically you'll end up with two persons for the same real person. Your copy, associated with your photos, and your partner's copy.

I'm a bit challenged on how to best solve sharing the actual person object, so both(all) users's photos will be assigned to the same person.

**Reply by risingPhil** [reply]
I am not burdened with any knowledge of the codebase nor database, but I suppose you could implement linking a Person entity to other Person entities in the database.

However, it goes beyond that though: When you're trying to show pictures from a person, you will also have to filter duplicate pictures. But perhaps this can be done in the view itself: if you would request the list of pictures from the server alongside their hashes, then I suppose you could make the client (browser/android/whatever) filter pictures with identical hashes.

The latter would also be a great feature for the general use case with shared libraries though: If both libraries contain the same picture, then I only would want to have it shown once. So if the view could do the filtering based on identical hashes, then that prevents the need to deduplicate the libraries themselves.

**Comment by TortugaTwo** [comment]
Hi, I'm very new to immich, but I love it!  I definitely need this feature, and I think this proposed above would be a great start.  

It sounds like the core issues is that images and face libraries are read only. I'm an idiot and don't really know what I'm talking about, but would it be possible to give an admin setting that would simply set all users to use the same face id library? Obviously, people would have some face IDs for people that aren't even in their photos, but for users, like me, we just want to use this program for our immediate family, so that's not a big deal.

**Comment by dgtal1** [comment]
This is what I also described under #9075. It would be very beneficial for the External Library to have settings like these. 

Altough I think the issue description is misleading or at least matches the description only partially. Could you please rename it so it reflects the description better? So not only faces, but maps + settings to control this? 

I think maybe when you change this it would capture focus more quickly + may reduce the number of duplicates when someone searches for feature requests prior creating another one.

**Comment by Str1atum** [comment]
For the beginning just a duplication of the maps "include partners" feature for people would be a great start

**Comment by wsidl** [comment]
Feels like there is a complication with the face id belonging to different owners, and less about the access to the metadata (like with partner photos on the map which are just point entities).

I also recognize merging face IDs is also not a solution because what if partner sharing is removed? You are now left with a face id being used by two owners which they both may recognize as separate individuals, let alone the underlying queries to now filter out results.

I do like @TortugaTwo's suggestion to include a toggle when setting up partner sharing, but I also see a need for a new relationship on a user's owned face ids and if they are the same as a face id owned by a shared user.

This keeps the relationship loose while still allowing queries to include the relationship in various queries. The only complexity is a need for a 1-to-many relationship as a user may have many sharing partners, and one user's face id relationship to their sharing partner's face id, the sharing partner may decide not to have that relationship.

**Comment by hakes-species-09** [comment]
One thought on a way solve this while preserving privacy could be to treat shared images the same as images uploaded by the account owner (while still pointing to the file held in the sharer's storage). Then face detection, recognition etc. could be run independently without risk of sharing recognition data that could be private to the sharer (i.e. derived from images not explicitly shared with the account owner). A downside is that each user would have, initially, to assign names, merge, hide faces etc.. again.

**Reply by freewill1361** [reply]
I would be fine with this as in interim solution, to tag partner faces and it be exclusive for my account. At least I will be able to search across all assets that I have access to.

My Tags <----- Shared Image ------> Owners Tags

**Comment by blue-kaleidoscope** [comment]
I am also looking for that feature. =) In the meantime I will continue with account sharing. Not that great though.

**Comment by sannidhyaroy** [comment]
+1. Also, since the owner and the partner might recognize a person in two different names, Immich would need to show appropriate person names for each user for shared assets.

**Comment by dna113p** [comment]
Is the only way to deal with this right now to login with the same account on both devices? Is there any reason not to do that?

**Reply by ttyridal** [reply]
Yes, and no - if you trust the other person, not beyond the general "sharing account is bad".

**Reply by hakes-species-09** [reply]
You cannot also have photos in your library that are unshared when you use the same account. E.g you may only want to share photos of a specific event with someone..

**Comment by krypknot** [comment]
We are in a situation where as a family we are moving away from Google to Immich.  This is a highly sought after feature - to allow us to have a single 'Google Import' account that then becomes a partner with each family member.  Sharing not only the picture but all associated metadata would be a huge win.

For now we are having to all share a single account to search for a person, location, etc.

**Comment by sproid-i** [comment]
YES! To me this feature will be the milestone for "Good enough" alternative software, and is going to be when I will purchase 
the server lic.. support purchase.

**Comment by xCaptainChris** [comment]
I’ll throw in my hat here. This is a huge limitation for me and my wife as well and I think we will be abandoning shared accounts and logging  both devices into admin account which I hate but I want to be able to find her pictures of my kids. As it sits I don’t believe I can even tag her photos with my own tags- let alone search the ones she’s already done.

**Reply by dna113p** [reply]
Exact same situation here. The term in the app is "partner" sharing which I think should just mean sharing everything.

**Reply by Peronia** [reply]
Yeah same situation here. 2 accounts and no duplicate detection and no face recognition is a step back. I'll keep the 2 accounts and live with a lot of dups and separated albums. I think this will come but nobody can (and will) say when.

**Comment by xttlegendapi** [comment]
this is super important!!!

**Comment by figadore** [comment]
I thought I did my due diligence, testing all the features of immich with my partner before doing all the work to switch us over to have it be our primary image solution (manually recreating every album, doing all the face recognition sorting, etc). Then I discovered that we missed this gap, and now I have to move us back. My partner is not happy.
We don't have any privacy or data concerns on our end, so it would be great to share all the face and other metadata about photos, even if we have to explicitly opt-in.

**Reply by gaoyichuan** [reply]
I agree on this, since most use case of Immich is a small family, where minimal privacy concern exists. I think Immich can add a "family mode" or similar thing, to simply allow server-wide face/album sharing, etc. Of course many warnings should be given prior to enable this mode.

This could simplify implementing of partner face sharing.

**Comment by Qhilm** [comment]
@figadore, out of curiosity, which product are you currently using allows this sharing of metadata?

**Comment by ragnarkarlsson** [comment]
+1 to this feature request. 

I think my partner and I would be ok if we simply were able to see each other's facial recognition / map location heatmap items, even if that meant we had two versions of "Child A" and "Child 1". We would likely name them the same and so a search would ideally allow us to select more than one person to view all photos matching that query.

**Comment by darkenbula** [comment]
+1 to this as well. we are so close to move to Immich full time and i would happily purchase a license if this feature was included.

**Comment by elronzo** [comment]
+1
As a family we all need to have the same view on our photo library. Including detected faces, of course.

**Comment by Tronnic** [comment]
+1 this is the one missing feature that holds us back to move to immich completely. I'm thinking about a way around this would be to NOT use the upload feature of immich, but use nextcloud instead and place our photos in different paths. This way we could use a single immich account and still have a seperated folder. But I dont know if it is the solution.

**Comment by deddc23efb** [comment]
I came here to support this feature request.  Not having this feature is creating all sorts of bad setups ranging from shared accounts to mass duplication.  Shared accounts means that the new multiple user feature isn't working and massive duplication is a huge waste of CPU and storage space.

**Comment by alextran1502** [comment]
This feature involves a big refactor to basically every queries in the codebase since it  related to how we structured the asset ownership's relationship, it is happening in the background and will take times. 

I will be locking the thread since we know what involves in the process and there have been no additional information besides us being reminded that the users want this feature.
